sdk/plugin/grpc_backend_client.go

```diff
@@ -280,3 +280,23 @@ func (b *backendGRPCPluginClient) Type() logical.BackendType {
 
 	return logical.BackendType(reply.Type)
 }
+
+func (b *backendGRPCPluginClient) Version() logical.VersionInfo {
+	reply, err := b.versionClient.Version(b.doneCtx, &logical.Empty{})
+	if err != nil {
+		if stErr, ok := status.FromError(err); ok {
+			if stErr.Code() == codes.Unimplemented {
+				return logical.EmptyVersion
+			}
+		}
+		b.Logger().Warn("Unknown error getting plugin version", "err", err)
```

Comment:
Something would have to go seriously wrong for this to happen, which makes me a bit nervous to only log it server-side and continue. I see `Type` does something similar above, but perhaps it's worth adding an error return value to our new interface signature before we first ship it so that we have an easier time if we have need for it at a later date. WDYT?

Reply:
I was trying to mimic `Type`'s relative safety -- worst case scenario, the `Version` will just return `""`. I can think of few reasons why this might fail, like if the plugin dies or is running too slowly.

(`Type` won't even log a warning if an error occurs. :))

I'm open to adding an error return though; I was just trying to keep it as simple as possible.

Reply:
Yeah it's definitely nice to have that simplicity. I was thinking a little of possible future changes to the plugin architecture, e.g. what if we wanted to support remote plugins where transport errors might become less unusual. It's perfectly possible the error will never be a particularly useful part of the API though, so I'm not 100% about it.